frontend/src/routes/[[lang=locale]]/candidate/(protected)/+page.svelte

```diff
@@ -6,22 +6,141 @@
+  });
+
+  let basicInfoQuestionsLeft: number | undefined;
+  nofUnansweredBasicInfoQuestions?.subscribe((value) => {
+    basicInfoQuestionsLeft = value;
+  });
+
+  let basicInfoFilled: boolean | undefined;
+  basicInfoFilledStore?.subscribe((value) => {
+    basicInfoFilled = value;
+  });
+
+  function getNextAction() {
+    if (!!basicInfoFilled && !!opinionQuestionsFilled) {
+      return {
+        title: $t('candidateApp.homePage.title'),
+        explanation: $t('candidateApp.homePage.description'),
+        tip: $t('candidateApp.homePage.tip'),
+        buttonTextBasicInfo: $t('candidateApp.homePage.basicInfoButtonView'),
+        buttonTextQuestion: $t('candidateApp.homePage.questionsButtonView'),
+        buttonTextPrimaryActions: $t('candidateApp.homePage.previewButton'),
+        href: $getRoute(Route.CandAppPreview)
+      };
+    } else if (!!basicInfoFilled && !opinionQuestionsFilled) {
+      return {
+        title: $t('candidateApp.homePage.greeting', {userName}),
+        explanation: $t('candidateApp.homePage.explanation'),
+        buttonTextBasicInfo: $t('candidateApp.homePage.basicInfoButtonEdit'),
+        buttonTextQuestion: $t('candidateApp.homePage.questionsButton'),
+        buttonTextPrimaryActions: $t('candidateApp.homePage.questionsButton'),
+        href: $getRoute(Route.CandAppQuestions)
+      };
+    } else {
+      return {
+        title: $t('candidateApp.homePage.greeting', {userName}),
+        explanation: $t('candidateApp.homePage.explanation'),
+        buttonTextBasicInfo: $t('candidateApp.homePage.basicInfoButton'),
+        buttonTextQuestion: $t('candidateApp.homePage.questionsButton'),
+        buttonTextPrimaryActions: $t('candidateApp.homePage.basicInfoButtonPrimaryActions'),
+        href: $getRoute(Route.CandAppProfile)
+      };
+    }
+  }
 </script>
 
 <!--Homepage for the user-->
 
-<BasicPage title={$t('candidateApp.homePage.greeting', {userName})}>
+<BasicPage title={getNextAction().title}>
+  <Warning display={!dataEditable} slot="note"
+    >{$t('candidateApp.homePage.editingNotAllowedNote')}
+    {#if !opinionQuestionsFilled || !basicInfoFilledStore}
+      <br />
+      <br />
+      {$t('candidateApp.homePage.editingNotAllowedPartiallyFilled')}
+    {/if}
+  </Warning>
   <p class="max-w-md text-center">
-    {$t('candidateApp.homePage.explanation')}
+    {getNextAction().explanation}
+    {#if !!basicInfoFilled && !!opinionQuestionsFilled}
+      <br />
+      <br />
+      {getNextAction().tip}
+    {/if}
   </p>
   <Button
-    variant="main"
-    slot="primaryActions"
-    text={$t('candidateApp.homePage.buttonText')}
-    icon="next"
-    href={$getRoute(Route.CandAppProfile)} />
+    text={getNextAction().buttonTextBasicInfo}
+    icon="profile"
+    iconPos="left"
+    href={$getRoute(Route.CandAppProfile)}>
+    {#if basicInfoQuestionsLeft && basicInfoQuestionsLeft > 0}
+      <InfoBadge text={String(basicInfoQuestionsLeft)} classes="-left-4 -top-4" />
```

Comment:
In TypeScript, I believe it is better practice to cast to string using `basicInfoQuestionsLeft as string`